File: init_pour_vous_actions.dart
import 'package:flutter/material.dart';
import 'lib/modules/vie_eglise/services/pour_vous_action_service.dart';
import 'lib/modules/vie_eglise/services/action_group_service.dart';
import 'lib/modules/vie_eglise/models/pour_vous_action.dart';
import 'lib/modules/vie_eglise/models/action_group.dart';

/// Script d'initialisation pour créer des actions "Pour vous" de démonstration
void main() async {
  print('🚀 Initialisation des actions "Pour vous"...');
  
  final actionService = PourVousActionService();
  final groupService = ActionGroupService();
  
  try {
    // Créer les groupes par défaut
    print('📁 Création des groupes par défaut...');
    await groupService.createDefaultGroups();
    
    // Attendre un peu pour que les groupes soient créés
    await Future.delayed(const Duration(seconds: 2));
    
    // Récupérer les groupes créés
    final groups = await groupService.getAllGroups().first;
    print('✅ ${groups.length} groupes créés');
    
    // Créer des actions de démonstration
    final demoActions = [
      PourVousAction(
        id: '',
        title: 'Prise de Rendez-vous',
        description: 'Prenez rendez-vous avec un pasteur ou un responsable',
        icon: Icons.calendar_today,
        iconCodePoint: Icons.calendar_today.codePoint.toString(),
        actionType: 'navigation',
        targetModule: 'rendez_vous',
        targetRoute: '/rendez_vous',
        isActive: true,
        order: 1,
        createdAt: DateTime.now(),
        updatedAt: DateTime.now(),
        color: '#4CAF50',
        category: 'Services',
      ),
      PourVousAction(
        id: '',
        title: 'Mur de Prière',
        description: 'Partagez vos demandes de prière avec la communauté',
        icon: Icons.favorite,
        iconCodePoint: Icons.favorite.codePoint.toString(),
        actionType: 'navigation',
        targetModule: 'mur_priere',
        targetRoute: '/mur_priere',
        isActive: true,
        order: 2,
        createdAt: DateTime.now(),
        updatedAt: DateTime.now(),
        color: '#E91E63',
        category: 'Spirituel',
      ),
      PourVousAction(
        id: '',
        title: 'Groupes de Maison',
        description: 'Rejoignez un groupe de maison près de chez vous',
        icon: Icons.home,
        iconCodePoint: Icons.home.codePoint.toString(),
        actionType: 'navigation',
        targetModule: 'groupes',
        targetRoute: '/groupes',
        isActive: true,
        order: 3,
        createdAt: DateTime.now(),
        updatedAt: DateTime.now(),
        color: '#FF9800',
        category: 'Communauté',
      ),
      PourVousAction(
        id: '',
        title: 'Bible en Ligne',
        description: 'Accédez à la Bible et aux outils d\'étude',
        icon: Icons.book,
        iconCodePoint: Icons.book.codePoint.toString(),
        actionType: 'navigation',
        targetModule: 'bible',
        targetRoute: '/bible',
        isActive: true,
        order: 4,
        createdAt: DateTime.now(),
        updatedAt: DateTime.now(),
        color: '#3F51B5',
        category: 'Spirituel',
      ),
      PourVousAction(
        id: '',
        title: 'Bénévolat',
        description: 'Participez aux activités de service de l\'église',
        icon: Icons.volunteer_activism,
        iconCodePoint: Icons.volunteer_activism.codePoint.toString(),
        actionType: 'navigation',
        targetModule: 'benevolat',
        targetRoute: '/benevolat',
        isActive: true,
        order: 5,
        createdAt: DateTime.now(),
        updatedAt: DateTime.now(),
        color: '#9C27B0',
        category: 'Services',
      ),
      PourVousAction(
        id: '',
        title: 'Contactez-nous',
        description: 'Envoyez un message à l\'équipe pastorale',
        icon: Icons.message,
        iconCodePoint: Icons.message.codePoint.toString(),
        actionType: 'form',
        targetModule: 'message',
        actionData: {
          'formType': 'contact',
          'recipient': 'pasteur@jubile.fr'
        },
        isActive: true,
        order: 6,
        createdAt: DateTime.now(),
        updatedAt: DateTime.now(),
        color: '#607D8B',
        category: 'Contact',
      ),
    ];
    
    print('📝 Création de ${demoActions.length} actions de démonstration...');
    
    for (final action in demoActions) {
      try {
        await actionService.createAction(action);
        print('✅ Action créée: ${action.title}');
      } catch (e) {
        print('❌ Erreur lors de la création de "${action.title}": $e');
      }
    }
    
    print('🎉 Initialisation terminée avec succès !');
    print('📱 Vous pouvez maintenant tester l\'onglet "Pour vous" dans l\'application');
    
  } catch (e) {
    print('❌ Erreur lors de l\'initialisation: $e');
  }
}

File: init_pour_vous_actions.js
const admin = require('firebase-admin');

// Initialiser Firebase Admin
try {
  const serviceAccountPath = './firebase-admin-key.json';
  const serviceAccount = require(serviceAccountPath);
  
  admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
    databaseURL: "https://hjye25u8iwm0i0zls78urffsc0jcgj-default-rtdb.firebaseio.com"
  });
} catch (error) {
  console.log('⚠️  Configuration Firebase Admin avec les credentials par défaut');
  admin.initializeApp({
    projectId: 'hjye25u8iwm0i0zls78urffsc0jcgj'
  });
}

const db = admin.firestore();

// Actions par défaut
const defaultActions = [
  {
    title: 'Prendre le baptême',
    description: 'Faire une demande de baptême',
    iconCodePoint: 57421, // Icons.water_drop
    actionType: 'form',
    isActive: true,
    order: 1,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#2196F3'
  },
  {
    title: 'Rendez-vous avec le pasteur',
    description: 'Prendre un rendez-vous personnel',
    iconCodePoint: 59441, // Icons.person_add
    actionType: 'navigation',
    targetModule: 'rendez_vous',
    isActive: true,
    order: 2,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#4CAF50'
  },
  {
    title: 'Rejoindre une équipe',
    description: 'Intégrer un groupe ou une équipe',
    iconCodePoint: 57778, // Icons.group_add
    actionType: 'navigation',
    targetModule: 'groupes',
    isActive: true,
    order: 3,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#FF9800'
  },
  {
    title: 'Requêtes de prière',
    description: 'Demander une prière ou prier pour d\'autres',
    iconCodePoint: 59143, // Icons.favorite
    actionType: 'navigation',
    targetModule: 'mur_priere',
    isActive: true,
    order: 4,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#E91E63'
  },
  {
    title: 'Poser une question au pasteur',
    description: 'Envoyer une question personnelle',
    iconCodePoint: 59648, // Icons.help_outline
    actionType: 'form',
    isActive: true,
    order: 5,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#9C27B0'
  },
  {
    title: 'Proposer une idée',
    description: 'Partager une suggestion ou idée',
    iconCodePoint: 59644, // Icons.lightbulb_outline
    actionType: 'form',
    isActive: true,
    order: 6,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#FFC107'
  },
  {
    title: 'Chanter un chant spécial',
    description: 'Proposer un chant pour le service',
    iconCodePoint: 59470, // Icons.music_note
    actionType: 'form',
    isActive: true,
    order: 7,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#673AB7'
  },
  {
    title: 'Informations sur l\'église',
    description: 'En savoir plus sur notre église',
    iconCodePoint: 59651, // Icons.info_outline
    actionType: 'form',
    isActive: true,
    order: 8,
    createdAt: admin.firestore.Timestamp.now(),
    updatedAt: admin.firestore.Timestamp.now(),
    color: '#607D8B'
  }
];

async function initializePourVousActions() {
  try {
    console.log('🔍 Vérification de la collection pour_vous_actions...');
    
    const snapshot = await db.collection('pour_vous_actions').limit(1).get();
    
    if (snapshot.empty) {
      console.log('📝 Collection vide, création des actions par défaut...');
      
      const batch = db.batch();
      
      defaultActions.forEach(action => {
        const docRef = db.collection('pour_vous_actions').doc();
        batch.set(docRef, action);
      });
      
      await batch.commit();
      console.log(`✅ ${defaultActions.length} actions créées avec succès`);
    } else {
      console.log('✅ Collection pour_vous_actions existe déjà avec des données');
    }
    
    // Statistiques
    const allDocs = await db.collection('pour_vous_actions').get();
    const activeDocs = await db.collection('pour_vous_actions').where('isActive', '==', true).get();
    
    console.log('\n📊 Statistiques:');
    console.log(`   Total actions: ${allDocs.size}`);
    console.log(`   Actions actives: ${activeDocs.size}`);
    console.log(`   Actions inactives: ${allDocs.size - activeDocs.size}`);
    
    // Test de la requête qui posait problème
    console.log('\n🧪 Test de la requête principale...');
    const testQuery = await db.collection('pour_vous_actions')
      .where('isActive', '==', true)
      .orderBy('order')
      .get();
    
    console.log(`✅ Requête réussie, ${testQuery.size} actions trouvées`);
    
    if (testQuery.size > 0) {
      console.log('\n📋 Actions disponibles:');
      testQuery.docs.forEach((doc, index) => {
        const data = doc.data();
        console.log(`   ${index + 1}. ${data.title} (ordre: ${data.order})`);
      });
    }
    
  } catch (error) {
    console.error('❌ Erreur:', error);
    
    if (error.code === 'failed-precondition') {
      console.log('\n💡 Suggestion: Vérifiez que les index Firebase sont déployés');
      console.log('   Commande: firebase deploy --only firestore:indexes');
    }
  }
}

// Exécution
initializePourVousActions()
  .then(() => {
    console.log('\n✅ Initialisation terminée');
    process.exit(0);
  })
  .catch(error => {
    console.error('❌ Erreur fatale:', error);
    process.exit(1);
  });